src/Parser/Paladin/Holy/Modules/Features/StatWeights.js

```diff
@@ -0,0 +1,401 @@
+    if (this.owner.byPlayer(event) || this.owner.byPlayerPet(event)) {
+      const healVal = new HealingValue(event.amount, 0, 0);
+      this._handleHealEvent(event, healVal);
+    }
+  }
+  on_removebuff(event) {
+    if (this.owner.byPlayer(event) || this.owner.byPlayerPet(event)) {
+      if (event.absorb) {
+        const healVal = new HealingValue(0, 0, event.absorb);
+        this._handleHealEvent(event, healVal);
+      }
+    }
+  }
+  on_beacon_heal(event, originalHeal) {
+    const spellInfo = getSpellInfo(originalHeal.ability.guid, originalHeal.ability.name);
+    const healVal = new HealingValue(event.amount, event.absorbed, event.overheal);
+    this._handleHeal(spellInfo, originalHeal, healVal);
+  }
+  _handleHealEvent(event, healVal) {
+    const spellInfo = getSpellInfo(event.ability.guid, event.ability.name);
+    this._handleHeal(spellInfo, event, healVal);
+  }
+  _handleHeal(spellInfo, event, healVal) {
+    // Most spells are counted in healing total, but some spells scale not on their own but 'second hand' from other spells
+    // I adjust them out of total healing to preserve some accuracy in the "Rating per 1%" stat.
+    // Good examples of multiplier spells are Leech and Velens.
+    if (!spellInfo.multiplier) {
+      this.totalAdjustedHealing += healVal.effective;
+    }
+
+    if (spellInfo.ignored) {
+      return;
+    }
+
+    this._leech(event, healVal);
+
+    if (spellInfo.multiplier) {
+      // Multiplier spells aren't counted for weights because they don't **directly** benefit from stat weights
+      return;
+    }
+
+    if (spellInfo.int) {
+      this._intellect(event, healVal);
+    }
+    if (spellInfo.crit) {
+      this._criticalStrike(event, healVal);
+    }
+    if (spellInfo.hasteHpct) {
+      this._hasteHpct(event, healVal);
+    }
+    if (spellInfo.mastery) {
+      this._mastery(event, healVal);
+    }
+    if (spellInfo.vers) {
+      this._versatility(event, healVal);
+    }
+  }
+  _leech(event, healVal) {
+    const spellId = event.ability.guid;
+
+    // We have to calculate leech weight differently depending on if we already have any leech rating.
+    // Leech is marked as a 'multplier' heal, so we have to check it before we do the early return below
+    const hasLeech = this.statTracker.currentLeechPercentage > 0;
+    if (hasLeech) {
+      // When the user has Leech we can use the actual Leech healing to accuractely calculate its HPS value without having to do any kind of predicting
+      if (!healVal.overheal && spellId === SPELLS.LEECH.id) {
+        this.totalOneLeech += healVal.effective / this.statTracker.currentLeechRating; // TODO: Make a generic method to account for base percentage
+      }
+    } else {
+      // Without Leech we will have to make an estimation so we can still provide the user with a decent value
+      if (this.playerHealthMissing > 0) { // if the player is full HP this would have overhealed.
+        const healIncreaseFromOneLeech = 1 / 23000;
```

Comment:
this should just be `this.statTracker.leechRatingPerPercent`

Reply:
updated